libraries/app/database_api.cpp

```diff
@@ -2061,43 +2061,55 @@ vector<blinded_balance_object> database_api_impl::get_blinded_balances( const fl
 //                                                                  //
 //////////////////////////////////////////////////////////////////////
 
-vector<withdraw_permission_object> database_api::get_withdrawals_giver(account_id_type account, object_id_type start, uint32_t limit)const
+vector<withdraw_permission_object> database_api::get_withdrawals_giver(account_id_type account, withdraw_permission_id_type start, uint32_t limit)const
 {
    return my->get_withdrawals_giver( account, start, limit );
 }
 
-vector<withdraw_permission_object> database_api_impl::get_withdrawals_giver(account_id_type account, object_id_type start, uint32_t limit)const
+vector<withdraw_permission_object> database_api_impl::get_withdrawals_giver(account_id_type account, withdraw_permission_id_type start, uint32_t limit)const
 {
    FC_ASSERT( limit <= 100 );
    vector<withdraw_permission_object> result;
 
-   auto withdraw_index_end = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().end();
-   auto withdraw_itr = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().lower_bound(boost::make_tuple(account, start));
-   while( withdraw_itr->withdraw_from_account == account && result.size() < limit && withdraw_itr != withdraw_index_end)
+   const auto& withdraw_idx = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>();
+   auto withdraw_index_end = withdraw_idx.end();
+   auto withdraw_itr = withdraw_idx.upper_bound(boost::make_tuple(account, start));
```

Comment:
After changed `start` to `withdraw_permission_id_type` and using `upper_bound` at same time, it would be unable to return the object whose id is `1.12.0` (although in production it's been removed long before).

Reply:
yes but i added an extra block for it 2 calls in the. do yo think we should remove it ?

Reply:
But it's ambiguous when `start` is `1.12.0` (want it to be returned or not). IMHO we can just change `start` to `object_id_type` so users can pass in `0.0.0` for example.

Reply:
And make the in-code document about `start` clearer.

Reply:
extended `start` doc line.